charts/consul/templates/api-gateway-controller-clusterrole.yaml

```diff
@@ -245,11 +245,21 @@ rules:
   - patch
   - update
 {{- if .Values.global.enablePodSecurityPolicies }}
-- apiGroups: ["policy"]
-  resources: ["podsecuritypolicies"]
-  resourceNames:
-    - {{ template "consul.fullname" . }}-api-gateway-controller
+- apiGroups:
+  - policy
+  resources:
+  - podsecuritypolicies
+  verbs:
+  - use
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - roles
+  - rolebindings
   verbs:
-    - use
+  - create
+  - get
+  - list
+  - watch
```

Comment:
I'm not a huge fan of allowing the API gateway controller to create `Roles` + `RoleBindings`; however, I don't see any other way to accomplish attaching a `PodSecurityPolicy` to each and every `Deployment` that the controller creates. Due to this, I've narrowed down the permissions as much as possible - note that it can't `update` existing resources - and only granting these permissions when `global.enablePodSecurityPolicies`.

Reply:
Thanks so much for the detail in the comment. I really appreciate it.